hoot-core/src/main/cpp/hoot/core/conflate/tile/TileUtils.cpp

```diff
@@ -184,78 +146,42 @@ void TileUtils::writeTilesToOsm(
 
   int randomTileIndex = -1;
   if (selectSingleRandomTile)
-  {
     randomTileIndex = TileUtils::getRandomTileIndex(tiles, randomSeed);
-  }
 
+  OsmMapPtr boundaryMap = tilesToOsmMap(tiles, nodeCounts, randomTileIndex, selectSingleRandomTile);
+
+  OsmMapWriterFactory::write(boundaryMap, outputPath);
+  OsmMapWriterFactory::writeDebugMap(boundaryMap, "osm-tiles");
+}
+
+OsmMapPtr TileUtils::tilesToOsmMap(
+  const std::vector<std::vector<geos::geom::Envelope>>& tiles,
+  const std::vector<std::vector<long>>& nodeCounts,
+  int randomTileIndex, const bool selectSingleRandomTile)
+{
   OsmMapPtr boundaryMap(new OsmMap());
   int bboxCtr = 1;
   for (size_t tx = 0; tx < tiles.size(); tx++)
   {
     for (size_t ty = 0; ty < tiles[tx].size(); ty++)
     {
-      // TODO: This code could possibly be replaced by GeometryUtils::createMapFromBounds.
-      const geos::geom::Envelope env = tiles[tx][ty];
-      const double circularError = ConfigOptions().getCircularErrorDefaultValue();
-
-      NodePtr lowerLeft(
-        new Node(
-          Status::Unknown1,
-          boundaryMap->createNextNodeId(),
-          geos::geom::Coordinate(env.getMinX(), env.getMinY()),
-          circularError));
-      boundaryMap->addNode(lowerLeft);
-      NodePtr upperRight(
-        new Node(
-          Status::Unknown1,
-          boundaryMap->createNextNodeId(),
-          geos::geom::Coordinate(env.getMaxX(), env.getMaxY()),
-          circularError));
-      boundaryMap->addNode(upperRight);
-      NodePtr upperLeft(
-        new Node(
-          Status::Unknown1,
-          boundaryMap->createNextNodeId(),
-          geos::geom::Coordinate(env.getMinX(), env.getMaxY()),
-          circularError));
-      boundaryMap->addNode(upperLeft);
-      NodePtr lowerRight(
-        new Node(
-          Status::Unknown1,
-          boundaryMap->createNextNodeId(),
-          geos::geom::Coordinate(env.getMaxX(), env.getMinY()),
-          circularError));
-      boundaryMap->addNode(lowerRight);
-
-      WayPtr bbox(new Way(Status::Unknown1, boundaryMap->createNextWayId(), circularError));
-      bbox->addNode(lowerLeft->getId());
-      bbox->addNode(upperLeft->getId());
-      bbox->addNode(upperRight->getId());
-      bbox->addNode(lowerRight->getId());
-      bbox->addNode(lowerLeft->getId());
-
-      // If you want any tags added here to be seen in the geojson output, update the select
-      // statement in writeTilesToGeoJson, if the field is already supported in osmconf.ini OR if
-      // its not supported in osmconfi.ini, add custom handling as a workaround (see related note
-      // in writeTilesToGeoJson).
-
-      // gdal will recognize any closed way with the boundary tag as a polygon (tags
-      // for features recognized as polys configurable in osmconf.ini), which is the type of
-      // output we want...so we have to output this tag (can be removed after the fact)
-      bbox->setTag("boundary", "task_grid_cell");
```

Comment:
The `boundary` tag isn't needed anymore because of the way `OsmGeoJsonWriter` works now.